app/src/main/java/com/iti/android/team1/ebuy/ui/category/view/CategoriesAdapter.kt

```diff
@@ -0,0 +1,54 @@
+package com.iti.android.team1.ebuy.ui.category.view
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.compose.ui.graphics.Color
+import androidx.core.content.ContextCompat
+import androidx.recyclerview.widget.RecyclerView
+import com.iti.android.team1.ebuy.R
+import com.iti.android.team1.ebuy.databinding.CategoryRowBinding
+import com.iti.android.team1.ebuy.model.pojo.Category
+
+class CategoriesAdapter ( val onCategoryBtnClick : (id:Long,title:String)->Unit ): RecyclerView.Adapter<CategoriesAdapter.CategoriesHolder>() {
+
+    private var btnIndex:Int=0
+    private var categories: List<Category> = emptyList()
+
+    class CategoriesHolder(val binding: CategoryRowBinding) : RecyclerView.ViewHolder(binding.root){
```

Comment:
`inline` class will help you to get a reference for the outer class and can handle the logic of the view holder here